### PR TITLE
jest-puppeteer-axe: migrate to @axe-core/puppeteer

### PR DESCRIPTION
**Pull request by Hypnosphi**
`axe-puppeteer` is deprecated and renamed to `@axe-core/puppeteer`, see https://github.com/dequelabs/axe-puppeteer#deprecated-axe-puppeteer

### CODE REVIEW

**Comment by Hypnosphi** [comment]
@gziolo can you please review?

**Comment by gziolo** [comment]
@Hypnosphi, I'm back from extended leave. I will have a look at this PR in the upcoming days, thanks for proposing those changes 👍

**Comment by gziolo** [comment]
One note, there are legitimate failures in e2e tests raised by the new version of axe API. The best way of moving forward would be to skip them for the time being and report a separate issue for them. We already maintain the list of disabled rules:

https://github.com/WordPress/gutenberg/blob/bd45cc90e4b84731b1a92b7a5c777a1258057e36/packages/e2e-tests/config/setup-test-framework.js#L189-L204

**Comment by gziolo** [comment]
There are two things that would make this PR complete:

1. `aria-required-children` and `aria-required-parent` are two rules that need to be disabled in the check for e2e tests.
2. An entry in the CHANGELOG (https://github.com/WordPress/gutenberg/blob/master/packages/jest-puppeteer-axe/CHANGELOG.md) file describing the change would be very useful as well.

File: packages/e2e-tests/config/setup-test-framework.js [hunk 1]
```diff
@@ -201,6 +201,8 @@ async function runAxeTestsForBlockEditor() {
 			'link-name',
 			'listitem',
 			'region',
+			'aria-required-children',
+			'aria-required-parent',
```
Comment (gziolo) [inline packages/e2e-tests/config/setup-test-framework.js:205]: I see that `frame-title` is another new rule added that causes one of the tests to fail. So:
```suggestion
			'aria-required-parent',
			'frame-title',
```
should do the trick.

File: packages/jest-puppeteer-axe/CHANGELOG.md [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 ## New Features
 
-- The `axe-puppeteer` dependency has been updated from requiring `^1.0.0` to requiring `^1.1.0`.
+- Migrated to [@axe-core/puppeteer](https://github.com/dequelabs/axe-core-npm/tree/develop/packages/puppeteer).
```
Comment (gziolo) [inline packages/jest-puppeteer-axe/CHANGELOG.md:7]:
```suggestion
- Migrated `axe-puppeteer` to its new package name  [@axe-core/puppeteer](https://github.com/dequelabs/axe-core-npm/tree/develop/packages/puppeteer).
```

File: packages/jest-puppeteer-axe/CHANGELOG.md [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 ## New Features
```
Comment (gziolo) [inline packages/jest-puppeteer-axe/CHANGELOG.md:5]: Unrelated but necessary for the release process :)

```suggestion
### New Features
```

**Review by gziolo — COMMENTED** [review]
I added some additional comments that should conclude this PR.

**Comment by gziolo** [comment]
There's still one e2e test failing:
```
FAIL packages/e2e-tests/specs/local/demo.test.js (6.605s)
29
  ● new editor state › should be immediately saveable
30

31
    expect(jest.fn()).not.toHaveErrored(expected)
32

33
    Expected mock function not to be called but it was called with:
34
    ["Failed to inject axe-core into frame (about:blank)"],["Failed to inject axe-core into frame (about:blank)"],["Failed to inject axe-core into frame (about:blank)"]
35

36
      at assertExpectedCalls (../jest-console/build/@wordpress/jest-console/src/index.js:36:4)
37
          at runMicrotasks (<anonymous>)
38
```

It's surprising to see it. Let me restart tests and debug locally if it doesn't help.

**Review by gziolo — APPROVED** [review]
I tested locally and I see the same e2e test failure on `master` branch. Let's get this in and explore it separately. 

Thank you again for the patch 💯

**Pull request merged**